src/traces/sunburst/calc.js

```diff
@@ -189,9 +189,10 @@ exports.calc = function(gd, trace) {
     var pullColor;
     var scaleColor;
     var colors = trace.marker.colors || [];
+    var hasColors = !!colors.length;
     trace._hasColorscale = hasColorscale(trace, 'marker');
```

Comment:
Why don't we need 

```js
trace._hasColorscale = hasColorscale(trace, 'marker', 'colors')
```

here?

Reply:
Is it OK to call `hasColorscale` once in the `default.js` and set `trace._hasColorscale` there?

Reply:
Good call. Done in https://github.com/plotly/plotly.js/pull/4242/commits/2e179940a459bad7208c3ed84393f4065ddcb849.

Reply:
Nicely done!